pages/writing.md

```diff
@@ -16,18 +16,59 @@
-* [running in VSCode](https://www.julia-vscode.org/docs/stable/userguide/runningcode/)
-* startup file
+First here's some content on the [REPL](https://docs.julialang.org/en/v1/stdlib/REPL/).
+It would be nice to link to the previous section by the fact that some people use it as a development environment making heavy use of `edit`.
+Primarily, though, this section should talk about (some of) the different modes, and give examples of how to use them.
+* Julian mode: Writing and running code.
+    Lots to say but none of it very interesting as most people will interact with the REPL primarily from an IDE or one of the other modes.
+* Help mode: Querying information about Julia objects --- to see their documentation, type information, and where they are defined --- and symbols --- to see how they are used or typed.
+    It would be nice to have an example of each of these things.
+* Pkg mode: Creating and managing packages and environments.
+    More detail in section below so no example needed, just link to that section.
+* Shell mode: Functions as a terminal inside Julia, you can also execude shell commands from Julian mode.
+    An example of this is warranted but the rest is nicely explained by the first clause.
+
+<!-- I believe Revise.jl makes more sense to talk about once the context of packages (particularly local packages) has been introduced. -->
+
+<!-- The VSCode is important to write in tandem with or after the Development environments part is written to avoid overlap. -->
+Most people develop Julia using a tool that allows for interactive development.
+(Short explanation on what "Interactive development" means unless already covered in IDE section)
+In VScode, you can [run code](https://www.julia-vscode.org/docs/stable/userguide/runningcode/) in the following useful ways:
+* For interactive development, the standard way to use Julia, you can use hotkeys to run selections, lines, sections, or whole files in the currently open REPL.
+    * Select a part of code to run only that part,
+    * Put your cursor inside or after a language construct, such as a definition or function call, to run the whole construct,
+    * Use the "run file in REPL" hotkey to do just that.
+* You can run entire scripts from a new Julia session each time, but this isn't great as it takes a while to start a new REPL each time.
+    To work with a (more or less) fresh workspace each time, you should bundle code into [local packages](#package-management).
+
+Some code, however, you want to run on startup, every time Julia is loaded.
+While VSCode has settings that allow you to specify startup flags, more powerful is its ability to automatically run a [startup file](https://docs.julialang.org/en/v1/manual/command-line-interface/#Startup-file).
+This allows you to, among other things, load commonly used utility packages (some of which we will discuss later in this blog post):
+```julia
+using Revise
+using OhMyREPL
+```
```

Comment:
This is my startup file, I think it's a good one. Perhaps we should only put Revise, and defer OhMyREPL to the esthetics section

Reply:
Given that both Revise and OMR appear after the startup section, maybe we can write something more abstract like
```julia
using ExamplePackage

ExamplePackage.do_startup_related_stuff()
```
and then write "an example of a startup.jl file can be found [here]()" which links to a startup.jl that the specific packages mentioned in this blog and shows more of what's possible:
```julia
# Load some packages
using Revise
using OhMyREPL

# Change some settings
ENV["EDITOR"] = "code" # Set the default editor to VSCode
colorscheme!("GruvboxDark") # Set the colorscheme
const PLOTS_DEFAULTS = Dict(:theme => :wong) # Set a default theme for plotting packages
```